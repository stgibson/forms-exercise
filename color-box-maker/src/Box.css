.Box {
  margin-bottom: 5px;
  display: flex;
  flex-direction: row;
}

.Box-button {
  margin-left: 10px;
  height: 20px;
}