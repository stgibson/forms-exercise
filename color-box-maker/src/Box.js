import React from "react";
import "./Box.css";

const Box = ({ id, width, height, color, removeBox }) => {
  return (
    <div
      id={ id }
      className="Box"
      data-testid={ `${width}-${height}-${color}` }
    >
      <div
        style={
          { width: `${width}px`, height: `${height}px`, backgroundColor: color }
        }
      >
      </div>
      <button className="Box-button" onClick={ removeBox }>X</button>
    </div>
  );
};

export default Box;