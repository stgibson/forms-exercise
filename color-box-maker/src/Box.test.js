import React from "react";
import Box from "./Box";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
  render(<Box id={ 1 } width={ 200 } height={ 200 } color="green" />);
});

it("matches snapshot", () => {
  const { asFragment } =
    render(<Box id={ 1 } width={ 200 } height={ 200 } color="green" />);
  expect(asFragment()).toMatchSnapshot();
});