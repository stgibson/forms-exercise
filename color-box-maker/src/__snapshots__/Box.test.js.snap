// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="200-200-green"
    id="1"
  >
    <div
      style="width: 200px; height: 200px; background-color: green;"
    />
    <button
      class="Box-button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
